app/adapter/web/src/component/pages/shared/GithubSigninButton.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { Component } from 'react';
+
+import githubLogo from './github.svg';
+
+interface Props {
+  signInLink: string;
+}
+
+export class GithubSigninButton extends Component<Props> {
+  render() {
+    return (
+      <a href={this.props.signInLink}>
+        <div className={'button github'}>
+          <img
+            alt={'Sign in with github account'}
+            className={'icon'}
+            src={githubLogo}
+          />{' '}
+          Sign In with Github
+        </div>
+      </a>
```

Comment:
Please extract common code used between `GithubSigninButton` and `GoogleSigninButton` and  into `SignInButton` component to encourage code reuse.

Reply:
I mean composition